### PR TITLE
[BUBO-92] [BUBO-94] My project findings flow

### PR DESCRIPTION
**Pull request by icedevera**
[[BUBO-92](https://vacuum.atlassian.net/browse/BUBO-92?atlOrigin=eyJpIjoiOTM2NDY5MTI4ZWZlNGM3OWFlNjIzN2Q1NmJlM2MyMzYiLCJwIjoiaiJ9)]

[[BUBO-94](https://vacuum.atlassian.net/browse/BUBO-94?atlOrigin=eyJpIjoiZDBlYjY1ODBjOGZhNDI2ZmI2NzdmNDRjZjUzNWU3NTIiLCJwIjoiaiJ9)]

My project past tab (only show vulnerabilities for FINISHED contests):
image

My project vulnerabilities (deduplicated findings):
image

My Project vulnerability all reports (deduplicated finding findings):
image

My project finding detail: 
image

My project Audit details (not finished):
image

My project audit details (finished):
image

[BUBO-92]: https://vacuum.atlassian.net/browse/BUBO-92?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[BUBO-94]: https://vacuum.atlassian.net/browse/BUBO-94?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/sections/projects/MyProjectVulnerabilityReport.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  }
+
+  return (
+    <>
+      <div className="px-24">
+        <div className="flex items-center gap-12">
+          <Button
+            variant="outline"
+            size="small"
+            className="flex gap-2 uppercase"
```
Comment (matejfalat) [inline src/components/sections/projects/MyProjectVulnerabilityReport.tsx:53]: Just a note, all Buttons and also Links inside Buttons are `uppercase` by default, so no need to specify it everywhere.

File: src/components/sections/projects/MyProjectVulnerabilityReport.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+            <div className="mb-12 flex flex-col gap-3">
+              <h3 className="text-bodyL text-purple-light">Uploaded files</h3>
+              <div className="grid grid-cols-5 gap-3">
+                {data.findingAttachments.length > 0 ? (
+                  data.findingAttachments.map((attachment) => (
+                    <Link
+                      key={attachment.attachmentUrl}
+                      href={attachment.attachmentUrl}
+                      className="flex items-center gap-2">
```
Comment (matejfalat) [inline src/components/sections/projects/MyProjectVulnerabilityReport.tsx:144]: Maybe `target="_blank"`?

File: src/components/sections/contest/ContestDetails/ContestInfo.tsx [hunk 3]
```diff
@@ -26,13 +31,24 @@ const ContestInfo = ({contest}: ContestInfoProps) => {
       <div className="mb-12 flex flex-col gap-3">
         <h3 className="text-bodyL text-purple-light">Scope definition</h3>
         {contest.filesInScope.map((fileUrl) => (
-          <Link href={contest.repoUrl} key={fileUrl} className="flex gap-3">
+          <Link href={fileUrl} key={fileUrl} className="flex gap-3">
```
[the remark this answers is not in the document]
Comment (icedevera) [reply src/components/sections/contest/ContestDetails/ContestInfo.tsx:34]: Ahh I see. I'll revert it back to `contest.repoUrl`

File: src/components/sections/contest/ContestDetails/ContestOverview.tsx [hunk 4]
```diff
@@ -59,7 +60,16 @@ const ContestOverview = ({contest, myProject}: ContestOverviewProps) => {
         )}
       </div>
       <div className="mt-12 grid grid-cols-4 grid-rows-2 gap-6">
-        {contest.endDate < new Date() ? (
+        {DateTime.fromJSDate(contest.startDate) < DateTime.now() &&
+        DateTime.fromJSDate(contest.endDate) > DateTime.now() &&
+        contest.status === ContestStatus.APPROVED ? (
```
Comment (matejfalat) [inline src/components/sections/contest/ContestDetails/ContestOverview.tsx:61]: You could consider extracting this into `isContestRunning` variable for readability.

**Review by matejfalat — APPROVED** [review]

File: src/server/actions/deduplicatedFinding/getDeduplicatedFinding.ts [hunk 5]
```diff
@@ -64,7 +64,7 @@ export const getDeduplicatedFindingsAction = async ({
   limit,
   offset = 0,
   sort = {
-    field: MyProjectVulnerabilitiesSorting.FOUND_BY,
+    field: MyProjectVulnerabilitiesSorting.SEVERITY,
```
Comment (matejfalat) [inline src/server/actions/deduplicatedFinding/getDeduplicatedFinding.ts]: Shouldn't the default sorting be also changed on client?

Comment (icedevera) [reply src/server/actions/deduplicatedFinding/getDeduplicatedFinding.ts]: Currently client is using undefined for sort params. Changed the prefetch call in `page` to be undefined as well 👍

**Pull request merged**